packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts

```diff
@@ -823,8 +854,8 @@ test('jsx', async () => {
   const result = await transformWithEsbuild(code, id)
   expect(await ssrTransformSimpleCode(result.code, '/foo.jsx'))
     .toMatchInlineSnapshot(`
-      "const __vite_ssr_import_0__ = await __vite_ssr_import__("react", {"importedNames":["default"]});
-      const __vite_ssr_import_1__ = await __vite_ssr_import__("foo", {"importedNames":["Foo","Slot"]});
+      "const __vite_ssr_import_1__ = await __vite_ssr_import__("foo", {"importedNames":["Foo","Slot"]});
+      const __vite_ssr_import_0__ = await __vite_ssr_import__("react", {"importedNames":["default"]});
```

Comment:
It seems we need https://github.com/Rich-Harris/magic-string/issues/180 to be solved, otherwise we cannot guarantee the order of the imports.

Reply:
See https://github.com/vitejs/vite/pull/16356#discussion_r1805953584